.github/workflows/update_gh_pages.yml

```diff
@@ -0,0 +1,43 @@
+# This workflow updates GitHub Pages on the repo it is run from.
+
+name: Update GitHub Pages
+
+on:
+  workflow_dispatch:  # Manually trigger. Colon is required.
+
+permissions:
+  contents: write  # For checkout and push.
+
+jobs:
+  update-github-pages:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+      with:
+        ref: master
+
+    - name: Setup Node
+      uses: actions/setup-node@v3
+      with:
+        node-version: 16
+
+    - name: NPM install
+      # Use CI so that we don't update dependencies in this step.
+      # TODO: remove --force flag from installing examples when examples dependencies are routinely tested
+      # Until then, don't block updating just because some dependencies are out of date.
+      run: |
+        npm ci
+        cd examples
+        npm ci --force
+        cd ..
+
+    - name: Deploy to GH Pages
+      env:
+        GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      # deploys to the current repo, so you can use this workflow from a fork to update your personal gh-pages
+      run: |
+        npm run deploy:prepare
+        git remote set-url origin https://git:${GH_TOKEN}@github.com/${GITHUB_REPOSITORY}.git
```

Comment:
Will this work if one doesn't already have a remote named `origin`?  (I don't.)

Reply:
Oh, I realise that this is may be less of a concern than I thought, since I'm guessing _the repo on the CI machine_ has an `origin`, and that's what's relevant here.

(It's my local repo on my laptop that does not—because I kept getting confused between `origin` and `upstream`, and so have named them `google` and `cpcallen`.)